### PR TITLE
Api/route cleanup

### PR DESCRIPTION
**Pull request by bobheadxi**
## :construction_worker: Purpose

Clean up route registration, with the goal of eventually reducing the complexity of our API by making the current complexity a bit more apparent

## :rocket: Changes

Liberal use of gin's Groups

## :warning: Breaking Changes

none (i hope)

### CODE REVIEW

File: api/api.go [hunk 1]
```diff
@@ -133,113 +132,161 @@ func (api *API) ListenAndServe(addr string, tls *TLSConfig) error {
...
+			new.POST("/network", api.createHostedIPFSNetworkEntryInDatabase)
+		}
+		ipfsRoutes := ipfsPrivate.Group("/ipfs")
+		{
+			ipfsRoutes.POST("/check-for-pin/:hash", api.checkLocalNodeForPinForHostedIPFSNetwork) // admin locked
+			ipfsRoutes.POST("/object-stat/:key", api.getObjectStatForIpfsForHostedIPFSNetwork)
+			ipfsRoutes.POST("/pin/:hash", api.pinToHostedIPFSNetwork)
+			ipfsRoutes.POST("/add-file", api.addFileToHostedIPFSNetwork)
+			ipfsRoutes.POST("/add-file/advanced", api.addFileToHostedIPFSNetworkAdvanced)
+			ipfsRoutes.POST("/publish/details", api.publishDetailedIPNSToHostedIPFSNetwork)
```
Comment (postables) [inline api/api.go]: This should be `/api/v1/ipfs-private/ipns/publish/details`

**Comment by postables** [comment]
Spotted a few possible issues, however I will take a look I don't think it's related to this PR directly.
```
[GIN-debug] redirecting request 301: /api/v1/ipfs/pins/ --> /api/v1/ipfs/pins
[GIN-debug] redirecting request 307: /api/v1/ipfs/add-file --> /api/v1/ipfs/add-file/
[GIN] 2018/10/11 - 01:11:50 | 400 |  2.543650637s |      10.10.10.2 | POST     /api/v1/ipfs/add-file/
https://{{api_url}}:6767/api/v1/ipfs-private/ipfs/pin/:hash
```

**Comment by bobheadxi** [comment]
@postables I can't seem to reproduce your issue - can I see the error?

File: api/api.go (hunk 1 above)
Comment (bobheadxi) [reply api/api.go]: resolved @ 81dee0e9a769ea31dd6887d05088775fbbb473eb

**Pull request merged**